### PR TITLE
[14.0][MIG] sale invoice auto deliver

### PR DESCRIPTION
**Pull request by xavier-bouquiaux**
@rousseldenis

### CODE REVIEW

File: sale_invoice_auto_deliver/tests/test_sale_invoice_auto_deliver.py [hunk 1]
```diff
@@ -36,11 +36,16 @@ def test_sale_invoice_auto_deliver_no_availability(self):
         Check exception raises
         """
         product_id = self.env.ref("product.product_product_11")
```
Comment (rousseldenis) [inline sale_invoice_auto_deliver/tests/test_sale_invoice_auto_deliver.py]: If creating product, this is not necessary anymore

File: sale_invoice_auto_deliver/wizards/sale_make_invoice_advance.py [hunk 2]
```diff
@@ -1,16 +1,21 @@
-# © 2016  Laetitia Gangloff, Acsone SA/NV (http://www.acsone.eu)
+# © 2021  Laetitia Gangloff, Acsone SA/NV (http://www.acsone.eu)
```
Comment (rousseldenis) [inline sale_invoice_auto_deliver/wizards/sale_make_invoice_advance.py]: Don't change this (originally created at that time). Maybe 2016-2021

File: sale_invoice_auto_deliver/wizards/sale_make_invoice_advance.py [hunk 3]
```diff
@@ -37,21 +41,41 @@ def create_invoices(self):
...
-                raise exceptions.Warning(
+                raise UserError(
                     _(
                         "You cannot automatic deliver this order, "
                         "some products are not available"
                     )
                 )
             else:
-                self.advance_payment_method = "all"
+                self.advance_payment_method = "all_auto"
```
Comment (rousseldenis) [inline sale_invoice_auto_deliver/wizards/sale_make_invoice_advance.py]: Why changing this ?

File: sale_invoice_auto_deliver/wizards/sale_make_invoice_advance.py (hunk 2 above)
Comment (xavier-bouquiaux) [reply sale_invoice_auto_deliver/wizards/sale_make_invoice_advance.py]: I put 2016 back everywhere

File: sale_invoice_auto_deliver/tests/test_sale_invoice_auto_deliver.py (hunk 1 above)
Comment (xavier-bouquiaux) [reply sale_invoice_auto_deliver/tests/test_sale_invoice_auto_deliver.py]: sorry -> i removed the line

File: sale_invoice_auto_deliver/wizards/sale_make_invoice_advance.py (hunk 3 above)
Comment (xavier-bouquiaux) [reply sale_invoice_auto_deliver/wizards/sale_make_invoice_advance.py]: The value added in the selection is all_auto
i don't understand why all here
may-be older odoo version were able to understand all as all_auto

**Comment by xavier-bouquiaux** [comment]
@rousseldenis 
could we merge this ?

**Review by rousseldenis — APPROVED** [review]
Code review. Could you squash commits with same message ?

**Comment by xavier-bouquiaux** [comment]
> Code review. Could you squash commits with same message ?

done

**Comment by xavier-bouquiaux** [comment]
@rousseldenis : i've just rebased, let me know if i can be of nay help

**Comment by rousseldenis** [comment]
> @rousseldenis : i've just rebased, let me know if i can be of nay help

Yes, some pings to help approving

**Pull request merged**